util/setuidgid.c

```diff
@@ -133,14 +134,20 @@ set_group(const char * groupname)
 
 	/* Find numerical GID. */
 	errno = 0;
-	if ((group_info = getgrnam(groupname)) == NULL) {
-		if (errno)
+	if ((group_info = getgrnam(groupname)) != NULL) {
+		gid = group_info->gr_gid;
+	} else {
+		if (errno) {
 			warnp("getgrnam(\"%s\")", groupname);
-		else
-			warn0("No such group: %s", groupname);
-		goto err0;
+			goto err0;
+		} else {
+			/* Treat it as a numeric ID. */
+			if (PARSENUM(&gid, groupname)) {
```

Comment:
Why the deep nesting?  It's perfectly valid to write

```
if (foo) {
    ...
} else if (bar) {
    ...
} else if (baz) {
    ...
}
```

all at one level of indentation.

Reply:
Biggest reason was the comment.  Both of these look wrong:
```
                if (errno) {
                        warnp("getgrnam(\"%s\")", groupname);
                        goto err0;
                } else if (PARSENUM(&gid, groupname)) {
                        /* Treat it as a numeric ID. */
                        warn0("No such group: %s", groupname);
                        goto err0;
                }
```
The above looks wrong because the "treat it as a numerid ID" comes **after** the attempt to do so.

Alternatively,
```
                if (errno) {
                        warnp("getgrnam(\"%s\")", groupname);
                        goto err0;
                        /* Treat it as a numeric ID. */
                } else if (PARSENUM(&gid, groupname)) {
                        warn0("No such group: %s", groupname);
                        goto err0;
                }
```
the above looks wrong because the "treat it as a numeric ID" is part of the code block related to an `errno` from `getgrnam()`.

Maybe it's "obvious" what the code is doing, and you don't think we need a comment?  If you explicitly say so, I'll go ahead and remove the comment, but I think it's useful.